openadapt/strategies/mixins/agents.py

```diff
@@ -0,0 +1,155 @@
+        run_prompt_template (`str`, *optional*):
+            Pass along your own prompt if you want to override the default template for the `run` method.
+        additional_tools ([`Tool`], list of tools or dictionary with tool values, *optional*):
+            Any additional tools to include on top of the default ones. If you pass along a tool with the same name as
+            one of the default tools, that default tool will be overridden. See test_agents.py for an example.
+    """
+
+    screenshots = []
+    action_events = []
+
+    def __init__(
+        self,
+        recording: Recording,
+        api_key,
+        model_name=MODEL_NAME,
+        chat_prompt_template=None,
+        run_prompt_template=None,
+        additional_tools=None,
+        screenshots=[],
+    ):
+        super().__init__(
+            MODEL_NAME,
+            api_key,
+            chat_prompt_template,
+            run_prompt_template,
+            additional_tools,
+        )
+        if recording is None:
+            logger.warning("recording invalid")
+        else:
+            self.recording = recording
+            self.screenshots = screenshots
+            self.action_events = get_events(recording)
+
+    def prompt(self, n=-1, debug=False, diffs=False):
+        """
+        This function prompts the agent with a message and an image, asking them to identify what the user is
+        doing and what text they see in the image.
+
+        Args:
+            n (int): number of action events to prompt for. -1 for all.
+            debug (bool): whether to save screenshots and diffs to disk
+            diffs (bool): whether to prompt with diffs [more experimental]
+
+        Returns:
+            bool: whether the prompt was successful
+
+        Preconditions:
+            - `self.recording` is not `None`
+        """
+
+        if self.recording is None:
+            raise ValueError("recording is invalid")
+        else:
+            debug_path = f"{config.DEBUG_DIR_NAME}/{self.recording.task_description}-{self.recording.timestamp}"
+
+        for idx, action_event in enumerate(self.action_events):
+            if n != -1 and idx >= n:
+                break
+
+            screenshot = action_event.screenshot
+            if screenshot is not None:
+                self.screenshots.append(screenshot)
+                screenshot.crop_active_window(action_event)
+                diff = display_event(action_event, diff=True) if diffs else None
+
+                if debug:
+                    # create debug directories if it doesn't exist
+                    if not os.path.exists(config.DEBUG_DIR_NAME):
+                        logger.info("creating debug directory")
+                        os.mkdir(config.DEBUG_DIR_NAME)
+
+                    if not os.path.exists(debug_path):
+                        logger.info("creating new directory for this timestamp")
+                        os.mkdir(debug_path)
```

Comment:
`os.makedirs()` instead of `os.mkdir` twice

Reply:
[60bcad1](https://github.com/OpenAdaptAI/OpenAdapt/pull/192/commits/60bcad1e290053cc349edf0f7eaf912298c96692)